make fail with LaTeX error: "kernel/command-already-defined"

On macOS 10.13.5, with TexLive 2018, make fail. I got the same error with a MacPorts TexLive installation and a MacTex one. Needed fonts installed in /Library/Fonts.

```bash
Machine:src julesvernes$ make
latexmk -shell-escape -interaction=nonstopmode -halt-on-error -norc -pdflatex="xelatex %O %S" -pdf -dvi- -ps- ctfp.tex; \
	mv ctfp.pdf ../category-theory-for-programmers.pdf
Latexmk: This is Latexmk, John Collins, 17 Jan. 2018, version: 4.55.
Latexmk: applying rule 'pdflatex'...
Rule 'pdflatex': Rules & subrules not known to be previously run:
   pdflatex
Rule 'pdflatex': The following rules & subrules became out-of-date:
      'pdflatex'
------------
Run number 1 of rule 'pdflatex'
------------
------------
Running 'xelatex  -shell-escape -interaction=nonstopmode -halt-on-error -recorder  "ctfp.tex"'
------------
This is XeTeX, Version 3.14159265-2.6-0.99999 (TeX Live 2018/MacPorts 2018.47642_1) (preloaded format=xelatex)
 \write18 enabled.
entering extended mode
(./ctfp.tex
LaTeX2e <2018-04-01> patch level 2
Babel <3.18> and hyphenation patterns for 46 language(s) loaded.
(./preamble.tex (/opt/local/share/texmf-texlive/tex/latex/base/book.cls
Document Class: book 2014/09/29 v1.4h Standard LaTeX document class
(/opt/local/share/texmf-texlive/tex/latex/base/bk11.clo))
(/opt/local/share/texmf-texlive/tex/latex/minted/minted.sty
(/opt/local/share/texmf-texlive/tex/latex/graphics/keyval.sty)
(/opt/local/share/texmf-texlive/tex/latex/oberdiek/kvoptions.sty
(/opt/local/share/texmf-texlive/tex/generic/oberdiek/ltxcmds.sty)
(/opt/local/share/texmf-texlive/tex/generic/oberdiek/kvsetkeys.sty
(/opt/local/share/texmf-texlive/tex/generic/oberdiek/infwarerr.sty)
(/opt/local/share/texmf-texlive/tex/generic/oberdiek/etexcmds.sty
(/opt/local/share/texmf-texlive/tex/generic/oberdiek/ifluatex.sty))))
(/opt/local/share/texmf-texlive/tex/latex/fvextra/fvextra.sty
(/opt/local/share/texmf-texlive/tex/latex/base/ifthen.sty)
(/opt/local/share/texmf-texlive/tex/latex/etoolbox/etoolbox.sty)
(/opt/local/share/texmf-texlive/tex/latex/fancyvrb/fancyvrb.sty
Style option: `fancyvrb' v2.7a, with DG/SPQR fixes, and firstline=lastline fix 
<2008/02/07> (tvz))
(/opt/local/share/texmf-texlive/tex/latex/upquote/upquote.sty
(/opt/local/share/texmf-texlive/tex/latex/base/textcomp.sty
(/opt/local/share/texmf-texlive/tex/latex/base/ts1enc.def)))
(/opt/local/share/texmf-texlive/tex/latex/lineno/lineno.sty))
(/opt/local/share/texmf-texlive/tex/latex/tools/calc.sty)
(/opt/local/share/texmf-texlive/tex/latex/tools/shellesc.sty)
(/opt/local/share/texmf-texlive/tex/latex/ifplatform/ifplatform.sty
(/opt/local/share/texmf-texlive/tex/generic/oberdiek/pdftexcmds.sty
(/opt/local/share/texmf-texlive/tex/generic/oberdiek/ifpdf.sty))
(/opt/local/share/texmf-texlive/tex/generic/oberdiek/catchfile.sty) (./ctfp.w18
)) (/opt/local/share/texmf-texlive/tex/generic/xstring/xstring.sty
(/opt/local/share/texmf-texlive/tex/generic/xstring/xstring.tex))
(/opt/local/share/texmf-texlive/tex/latex/framed/framed.sty)
(/opt/local/share/texmf-texlive/tex/latex/float/float.sty))
(/opt/local/share/texmf-texlive/tex/latex/fontspec/fontspec.sty
(/opt/local/share/texmf-texlive/tex/latex/l3packages/xparse/xparse.sty
(/opt/local/share/texmf-texlive/tex/latex/l3kernel/expl3.sty
(/opt/local/share/texmf-texlive/tex/latex/l3kernel/expl3-code.tex)
(/opt/local/share/texmf-texlive/tex/latex/l3kernel/l3xdvipdfmx.def)))
(/opt/local/share/texmf-texlive/tex/latex/fontspec/fontspec-xetex.sty
(/opt/local/share/texmf-texlive/tex/latex/base/fontenc.sty
(/opt/local/share/texmf-texlive/tex/latex/base/tuenc.def))
(/opt/local/share/texmf-texlive/tex/latex/fontspec/fontspec.cfg)))
(/opt/local/share/texmf-texlive/tex/latex/newtx/newtxmath.sty
`newtxmath' v1.53, 2017/12/14 Math macros based on txfonts (msharpe)
(/opt/local/share/texmf-texlive/tex/latex/amsmath/amsmath.sty
For additional information on amsmath, use the `?' option.
(/opt/local/share/texmf-texlive/tex/latex/amsmath/amstext.sty
(/opt/local/share/texmf-texlive/tex/latex/amsmath/amsgen.sty))
(/opt/local/share/texmf-texlive/tex/latex/amsmath/amsbsy.sty)
(/opt/local/share/texmf-texlive/tex/latex/amsmath/amsopn.sty))
(/opt/local/share/texmf-texlive/tex/generic/ifxetex/ifxetex.sty)
(/opt/local/share/texmf-texlive/tex/latex/xkeyval/xkeyval.sty
(/opt/local/share/texmf-texlive/tex/generic/xkeyval/xkeyval.tex
(/opt/local/share/texmf-texlive/tex/generic/xkeyval/xkvutils.tex)))
(/opt/local/share/texmf-texlive/tex/latex/oberdiek/centernot.sty)
(/opt/local/share/texmf-texlive/tex/generic/kastrup/binhex.tex))
(/opt/local/share/texmf-texlive/tex/latex/unicode-math/unicode-math.sty
(/opt/local/share/texmf-texlive/tex/latex/unicode-math/unicode-math-xetex.sty
(/opt/local/share/texmf-texlive/tex/latex/l3packages/l3keys2e/l3keys2e.sty)
(/opt/local/share/texmf-texlive/tex/latex/filehook/filehook.sty)
(/opt/local/share/texmf-texlive/tex/latex/base/fix-cm.sty
(/opt/local/share/texmf-texlive/tex/latex/base/ts1enc.def))
(/opt/local/share/texmf-texlive/tex/latex/unicode-math/unicode-math-table.tex)

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
!
! LaTeX error: "kernel/command-already-defined"
! 
! Control sequence \not= already defined.
! 
! See the LaTeX3 documentation for further information.
! 
! For immediate help type H <return>.
!...............................................  
                                                  
l.3853 \NewNegationCommand { = }    { \neq   }
                                              
No pages of output.
Transcript written on ctfp.log.
=== TeX engine is 'XeTeX'
Latexmk: Log file says no output from latex
Latexmk: For rule 'pdflatex', no output was made
Latexmk: Errors, so I did not complete making targets
Collected error summary (may duplicate other messages):
  pdflatex: Command for 'pdflatex' gave return code 1
      Refer to 'ctfp.log' for details
Latexmk: Use the -f option to force complete processing,
 unless error was exceeding maximum runs of latex/pdflatex.
mv: rename ctfp.pdf to ../category-theory-for-programmers.pdf: No such file or directory
make: *** [ctfp.pdf] Error 1
```

--- a/shell.nix
+++ b/shell.nix
@@ -40,6 +40,7 @@ mkShell {
         chngcntr
         ifplatform
         xstring
+        minifp
         enumitem;
       })
     gnumake

--- a/shell.nix
+++ b/shell.nix
@@ -1,13 +1,5 @@
 # nix-shell --pure shell.nix --command 'cd src; make'
-let
-  rev = "4df3426f5a5e78cef4835897a43abd9e2a092b74";
-  sha256 = "05k5mssiqxffxi45mss9wjns6k76i248rpasa48akdcriry1mp63";
-  nixpkgs = builtins.fetchTarball {
-    inherit sha256;
-    url = "https://github.com/NixOS/nixpkgs-channels/archive/${rev}.tar.gz";
-  };
-in
-{ pkgs ? import nixpkgs {} }: with pkgs;
+{ pkgs ? import <nixpkgs> {} }: with pkgs;
 
 mkShell {
   FONTCONFIG_FILE = makeFontsConf { fontDirectories = [ inconsolata-lgc libertine libertinus ]; };

--- a/src/preamble.tex
+++ b/src/preamble.tex
@@ -34,6 +34,11 @@
 \usepackage[libertine]{newtxmath}
 \usepackage{amsmath}
 % Needed to display additional math unicode symbols (like double-colon)
+
+% https://github.com/wspr/unicode-math/issues/469
+\expandafter\let\csname not=\endcsname\relax
+\expandafter\let\csname not<\endcsname\relax
+\expandafter\let\csname not>\endcsname\relax
 \usepackage{unicode-math}
 \setmathfont{Libertinus Math}
 

